app/assets/javascripts/discourse/helpers/category-link.js.es6

```diff
@@ -38,6 +39,12 @@ export function categoryBadgeHTML(category, opts) {
   )
     return "";
 
+  if (opts.recursive) {
```

Comment:
Can you add a way to prevent infinite recursion? Eg. probably based on the site setting?